geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionConfigImpl.java

```diff
@@ -1544,12 +1548,32 @@ void setApiProps(Properties apiProps) {
     }
   }
 
-  private static boolean specialPropName(String propName) {
-    return propName.equalsIgnoreCase(CLUSTER_SSL_ENABLED)
-        || propName.equals(SECURITY_PEER_AUTH_INIT) || propName.equals(SECURITY_PEER_AUTHENTICATOR)
-        || propName.equals(LOG_WRITER_NAME) || propName.equals(DS_CONFIG_NAME)
-        || propName.equals(SECURITY_LOG_WRITER_NAME) || propName.equals(LOG_OUTPUTSTREAM_NAME)
-        || propName.equals(SECURITY_LOG_OUTPUTSTREAM_NAME);
+  /**
+   * internal configuration properties not in ConfigurationProperties but that may be in
+   * properties fed to a DistributionConfigImpl constructor.
+   */
+  static Set<String> internalPropertyNames = new HashSet<>(Arrays.asList(
+      LOG_WRITER_NAME, DS_CONFIG_NAME,
+      DS_QUORUM_CHECKER_NAME, DS_RECONNECTING_NAME,
+      SECURITY_LOG_WRITER_NAME, LOG_OUTPUTSTREAM_NAME,
+      SECURITY_LOG_OUTPUTSTREAM_NAME));
+
```

Comment:
could we get configuration properties from the DistributionConfig interface's fields using reflection to build this list?